File: backend/app/schemas/simulation.py
from datetime import datetime

from pydantic import BaseModel, Field

from backend.app.schemas.geometry import GeoJSONGeometry


class SimulationRequest(BaseModel):
    region_id: str
    flood_level: float = Field(ge=0, description="Flood level in metres — see docs/DATA_FORMATS.md §3")


class SimulationResponse(BaseModel):
    scenario_id: str
    region_id: str
    flood_level: float
    status: str  # pending | running | completed | failed — docs/DATA_FORMATS.md §4
    flooded_area: float | None = None  # km²
    population_affected: int | None = None
    buildings_affected: int | None = None  # null unless optional `buildings` table adopted
    roads_affected_count: int | None = None  # always a count, never a length
    hospitals_affected: int | None = None
    result_geometry: GeoJSONGeometry | None = None
    created_at: datetime

File: frontend/src/types/simulation.ts
import { GeoJSONGeometry } from './geojson';

export interface SimulationRequest {
  region_id: string;
  flood_level: number; // in meters (>= 0)
}

export type SimulationStatus = 'pending' | 'running' | 'completed' | 'failed';

export interface SimulationResponse {
  scenario_id: string;
  region_id: string;
  flood_level: number;
  status: SimulationStatus;
  flooded_area?: number | null; // km²
  population_affected?: number | null;
  buildings_affected?: number | null;
  roads_affected_count?: number | null; // Always segment count, never km
  hospitals_affected?: number | null;
  result_geometry?: GeoJSONGeometry | null;
  created_at?: string;
}
